apps/notifications/src/panel/rest-client/index.js

```diff
@@ -211,17 +205,17 @@ function getNotes() {
 
 		store.dispatch( actions.ui.loadedNotes() );
 
-		const oldNotes = getAllNotes( store.getState() ).map( property( 'id' ) );
-		const newNotes = data.notes.map( property( 'id' ) );
-		const notesToRemove = difference( oldNotes, newNotes );
+		const oldNotes = getAllNotes( store.getState() ).map( ( n ) => n.id );
+		const newNotes = data.notes.map( ( n ) => n.id );
+		const notesToRemove = oldNotes.filter( ( old ) => ! newNotes.includes( old ) );
 
 		notesToRemove.length && store.dispatch( actions.notes.removeNotes( notesToRemove ) );
 		store.dispatch( actions.notes.addNotes( data.notes ) );
 
 		// Store id/hash pairs for now until properly reduxified
 		// this is used as a network optimization to quickly determine
 		// changes without downloading all the data
-		this.noteList = data.notes.map( ( note ) => pick( note, [ 'id', 'note_hash' ] ) );
+		this.noteList = data.notes.map( ( note ) => ( { id: note.id, note_hash: note.note_hash } ) );
```

Comment:
Destructuring can make it shorter here:

```suggestion
		this.noteList = data.notes.map( ( { id, note_hash } ) => ( { id, note_hash } ) );
```